modules/custom/az_person/az_person_profile_import/migrations/az_person_profile_import.yml

```diff
@@ -0,0 +1,101 @@
+id: az_person_profile_import
+label: Profiles Integration
+migration_tags:
+  - Profiles Integration
+source:
+  plugin: url
+  data_fetcher_plugin: http
+  data_parser_plugin: az_person_profile_import_json
+  urls: []
+  item_selector: /
+  ids:
+    netid:
+      type: string
+  constants:
+    NAME: 'name'
+
+  fields:
+    -
+      name: netid
+      selector: 'Person/netid'
+    -
+      name: surname
+      selector: 'Person/surname'
+    -
+      name: givenname
+      selector: 'Person/givenname'
+    -
+      name: email
+      selector: 'Person/email'
+    -
+      name: biography
+      selector: 'Bio/desc'
+    -
+      name: titles
+      selector: 'Titles'
+    -
+      name: phone
+      selector: 'Person/phone'
+    -
+      name: degrees
+      selector: 'Degrees'
+
+process:
+  type:
+    plugin: default_value
+    default_value: az_person
+  field_az_fname: givenname
+  field_az_lname: surname
+  field_az_email: email
+  field_az_body/value: biography
+  field_az_body/format:
+    plugin: default_value
+    default_value: az_standard
+  field_az_titles:
+    plugin: sub_process
+    source: titles
+    process:
+      value: desc
+  field_az_degrees:
+    -
+      plugin: sub_process
+      source: degrees
+      include_source: true,
+      source_key: original_source
+      process:
+        0: 'properties/degree_title'
+        1:
+          -
+            plugin: callback
+            callable: array_column
+            unpack_source: true
+            source:
+              - institution
+              - 'original_source/constants/NAME'
+          -
+            plugin: concat
+            delimiter: ', '
+    -
+      plugin: sub_process
+      process:
+        value:
+          -
+            plugin: concat
+            source:
+              - 0
+              - 1
+            delimiter: ', '
```

Comment:
After some research on a profile integration on a departmental site, I found they were importing degrees so that they were formatted like:

```html
<ul>
  <li>Bachelor of Science, Computer Science, University of Example, 2020</li>
</ul>

```

The migration appears to fomats degrees like:

```html
<ul>
  <li>Bachelor of Science, University of Example</li>
</ul>

```

Not sure  if it helps, but here is the logic from the departmental importer.

Degrees are sorted by year

```
  private function degree_sort_cmp($a, $b)
  {
    $a_degree_conferred_year = intval(isset($a->CONFERRED_DEGREE->year) ? $a->CONFERRED_DEGREE->year : 0);
    $b_degree_conferred_year = intval(isset($b->CONFERRED_DEGREE->year) ? $b->CONFERRED_DEGREE->year : 0);
    if ($a_degree_conferred_year == $b_degree_conferred_year)
    {
      return 0;
    }
    return ($a_degree_conferred_year < $b_degree_conferred_year) ? 1 : -1;
  }

```
Degrees are formatted
```
  public function getDegrees()
  {
    $html = "";
    if (!empty($this->degrees))
    {
      $html .= "<ul>";
      foreach ($this->degrees as $degree) {
        $html .= "<li>" . $degree->properties->degree_type;
        if (isset($degree->properties->discipline)) {
          $html .= ", " . $degree->properties->discipline;
        }
        if (isset($degree->institution[0]->name)) {
          $html .= ", " . $degree->institution[0]->name;
        }
        if (isset($degree->CONFERRED_DEGREE->year)) {
          $html .= ", " . $degree->CONFERRED_DEGREE->year;
        }
        $html .= "</li>";

      }
      $html .= "</ul>";
    }
    return $html;
  }

```

Reply:
Here is what the degrees field looks like for reference
<img width="1088" alt="image" src="https://github.com/az-digital/az_quickstart/assets/1023167/1b11529d-1bd1-4ea7-a330-b248160e1106">

Reply:
Agreed, I think we should add this.